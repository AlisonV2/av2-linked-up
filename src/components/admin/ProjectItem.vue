<template>
  <div class="card text-center">
    <div class="card-header title">
      {{ project.title }}
    </div>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'client'">{{ project.artistName }}</h6>
    <h6 class="card-subtitle mb-2 text-muted" v-if="role === 'artist'">{{ project.clientName }}</h6>
    <p class="card-text">{{ project.description }}</p>
    <div class="card-footer text-muted">
      {{ createdAt }}
    </div>
  </div>
</template>

<script>
import { format } from 'date-fns';

/**
 * @exports ProjectItem
 * @type {Component}
 * @requires Sentry
 * @vue-data {array} projects
 */
export default {
  name: 'ProjectItem',
  props: ['project', 'time'],
  data() {
    return {
      createdAt: '',
      role: '',
    };
  },
  created() {
    this.formatDate();
    this.getUserRole();
  },
  methods: {
    formatDate() {
      this.createdAt = format(this.time.toDate(), 'dd/MM/yyyy');
    },
    async getUserRole() {
      await this.$store.dispatch('getUserRole').then(() => {
        this.role = this.$store.getters.getUserRole;
      });
    },
  },
};
</script>
