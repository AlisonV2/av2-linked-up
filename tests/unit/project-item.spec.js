import { shallowMount } from '@vue/test-utils';
import { createStore } from 'vuex';
import ProjectItem from '@/components/admin/ProjectItem.vue';

const actions = {
  getUserRole: jest.fn(),
};

const getters = {
  getUserRole: jest.fn(),
};

const store = createStore({
  actions,
  getters,
});

const clientConfig = {
  data() {
    return {
      role: 'client',
      status: 'Pending',
      project: {
          title: 'Title',
          artistName: 'Artist',
          clientName: 'Client',
          description: 'Description',
          createdAt: new Date().toString()
      },
      isArtist: false,
    };
  },
  global: {
    plugins: [store],
  },
};

/**
 * @module ProjectItemTest
 */
describe('ProjectItem.vue', () => {
  it('ProjectItem Snapshot', () => {
    const component = shallowMount(ProjectItem, clientConfig);
    expect(component.element).toMatchSnapshot();
  });

  it('getUserRole method should be called', async () => {
    const component = shallowMount(ProjectItem, clientConfig);
    ProjectItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(component.vm.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole action should be called', async () => {
    const component = shallowMount(ProjectItem, clientConfig);
    ProjectItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(actions.getUserRole).toHaveBeenCalled;
  });

  it('getUserRole getter should be called', async () => {
    const component = shallowMount(ProjectItem, clientConfig);
    ProjectItem.created.call(component.vm);
    await component.vm.$nextTick();
    expect(getters.getUserRole).toHaveBeenCalled;
  });
});
