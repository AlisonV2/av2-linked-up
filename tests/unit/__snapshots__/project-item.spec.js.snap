// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectItem.vue ProjectItem Snapshot 1`] = `
<div
  class="card text-center client-card"
>
  <div
    class="card-header title"
  >
    Title 
    <span
      class="project-bullet project-pending"
    >
      â—‰
    </span>
    <h6
      class="card-subtitle mb-2 text-muted"
    >
       Project sent to Artist
    </h6>
    <!--v-if-->
  </div>
  <p
    class="card-text"
  >
    Description
  </p>
  <div
    class="card-footer text-muted"
  >
    Thu Nov 18 2021 08:27:13 GMT+0100 (Central European Standard Time)
  </div>
</div>
`;
